website/integrations: jellyfin: add OIDC configuration (#9538)
goauthentik/authentik: website/integrations/services/jellyfin/index.md

    - Roles: roles to look for when authorizing access (should be done through authentik instead)
    - Admin Roles: roles to look for when giving admin privilege
    - Role Claim: `groups`

10. Hit **Save** at the bottom.
11. On the left side now click the **General** under dashboard and go to **Branding**.
12. In the login disclaimer put this code and making sure to change the url at the top:

```
<form action="https://jellyfin.company.com/sso/OID/start/authentik">
  <button class="raised block emby-button button-submit">
    Sign in with SSO
  </button>
</form>
```

13. In the Custom CSS code also add this:

```
a.raised.emby-button {
    padding:0.9em 1em;
    color: inherit !important;
}

.disclaimerContainer{
    display: block;
}
```

14. Click **Save** at the bottom & restart the server.
15. When you are signed out you should now see a **Sign in with SSO** button.

:::note
If you have problems check your logs which are under the **Administration** -> **Dashboard** then "logs" and will be near the bottom (most likely) with `Jellyfin.Plugin.SSO_Auth.` as the start of the lines you are looking for.
:::

### authentik Configuration

> To use the role claim within Jellyfin we will have to add a custom property. This is not needed but does allow admins to be set for SSO users automatically. It can also deny access if they don't have a role, but this should be done though authentik instead.

Reviewer: Pretty sure this is inexact, as I have this setup at home without a custom mapping. I'll have to double check.

Author: Yeah I just tested this and it looks like that's true. I'll updated the documentation soon.

1. Click "Save"
2. Logout, and login with a LDAP user. Username **must** be used, logging in with email will not work.

> ### Authentik Configuration

Author: Interesting, I  just noticed that!

No additional authentik configuration needs to be configured. Follow the LDAP outpost instructions to create an LDAP outpost and configure access via the outpost

## OIDC Configuration

> ### authentik Configuration

Reviewer 2: Consider switching the order of authentik and Jellyfin (on line 136) so they match the order in the LDAP section above... This would also help lessen the initial confusion of having two H3s that both say "authentik configuration". ;)

Author: Cool :)

> To use the role claim within Jellyfin we will have to add a custom property. This is not needed but does allow admins to be set for SSO users automatically. It can also deny access if they don't have a role, but this should be done though Authentik instead.

Reviewer 2: ```suggestion
To use the role claim within Jellyfin we will have to add a custom property. This is not needed but does allow admins to be set for SSO users automatically. It can also deny access if they don't have a role, but this should be done though authentik instead.
```

> 1. Go to _Customization_ then _Property Mappings_

Reviewer 2: ```suggestion
1. In the Admin interface, go to **Customization -> Property Mappings**.
```

> 2. Create a _Scope Mapping_

Reviewer 2: ```suggestion
2. Create a **Scope Mapping**.
```

3. Assign these values:
    - name: `Group Membership`
    - scope name: `groups`
    - description: `See which groups you belong to`
    - expression: `return [group.name for group in user.ak_groups.all()]`

![](./jellyfin_mapping.png)

**Provider Settings**

> In authentik under _Providers_, create an _OAuth2/OpenID Provider_ with these settings:

Reviewer 2: ```suggestion
In authentik under **Providers**, create an OAuth2/OpenID Provider with these settings:
```

-   Name: `jellyfin`
-   Redirect URI: `https://jellyfin.company.com/sso/OID/redirect/authentik`
-   Signing Key: Select any available key

> For the Group Membership we just created go into _Advanced Protocol Settings_ and add that Group Membership to the selected.

Reviewer 2: ```suggestion
For the Group Membership we just created go into **Advanced Protocol Settings** and add that Group Membership to the selected.
```

Reviewer 2: @Redlonghead to what exactly are they adding the Group Membership to...? "To the selcted..."?

Author: My bad they are adding the scope called "Group Membership" to the scope under **Advanced Protocol Settings**. I had it sown in the photo but didn't specify it in the actual documentation

![](./jellyfin_scopes.png)

Make sure to grab the Client ID and the Client Secret!

:::note

> The last part of the URI is the name you use when making the provider in Jellyfin so make sure they are the same

Reviewer 2: ```suggestion
The last part of the URI is the name you use when making the provider in Jellyfin so make sure they are the same.
```

:::

**Application Settings**

> Create an application which uses `jellyfin`. Optionally apply access restrictions to the application.

Reviewer 2: ```suggestion
Create an application that uses `jellyfin`. Optionally apply access restrictions to the application.
```

1. Navigate to your Jellyfin installation and log in with the admin account or currently configured local admin.
2. Open the administrator dashboard and go to the "Plugins" section.
3. Then click the "Repositories" section at the top and add the below repository with the name of SSO-Auth

```
https://raw.githubusercontent.com/9p4/jellyfin-plugin-sso/manifest-release/manifest.json
```

4. Click the "Catalog" tab on top and install the SSO-Auth with the most recent version.
5. Restart the Jellyfin server.
6. Go back to the plugin tab.
7. Click the SSO-Auth plugin.
8. Fill out the Add / Update Provider Configuration:

    - Name of OID Provider: `authentik`
    - OID Endpoint: `https://authentik.company.com/application/o/jellyfin/.well-known/openid-configuration`
    - OpenID Client ID: ClientID from provider
    - OID Secret: Client Secret from provider
    - Enabled: **CHECKED**
    - Enable Authorization by Plugin: **CHECKED**

9. If you want to use the role claim then also fill out these:

    - Roles: roles to look for when authorizing access (should be done through authentik instead)
    - Admin Roles: roles to look for when giving admin privilege
    - Role Claim: `groups`

> 10. Hit save at the bottom.

Reviewer 2: ```suggestion
10. Click **Save** at the bottom.
```

> 11. On the left side now click the "General" under dashboard and goto "Branding".

Reviewer 2: ```suggestion
11. On the left side now click the "General" under dashboard and go to "Branding".
```

12. In the login disclaimer put this code and making sure to change the url at the top:

```
<form action="https://jellyfin.company.com/sso/OID/start/authentik">
  <button class="raised block emby-button button-submit">
    Sign in with SSO
  </button>
</form>
```

13. In the Custom CSS code also add this:

```
a.raised.emby-button {
    padding:0.9em 1em;
    color: inherit !important;
}

.disclaimerContainer{
    display: block;
}
```

> 14. Click save at the bottom & restart the server.

Reviewer 2: ```suggestion
14. Click **Save** at the bottom & restart the server.
```

9. If you want to use the role claim then also fill out these:

    - Roles: roles to look for when authorizing access (should be done through authentik instead)
    - Admin Roles: roles to look for when giving admin privilege
    - Role Claim: `groups`

10. Hit **Save** at the bottom.
11. On the left side now click the **General** under dashboard and go to **Branding**.
12. In the login disclaimer put this code and making sure to change the url at the top:

```
<form action="https://jellyfin.company/sso/OID/start/authentik">
  <button class="raised block emby-button button-submit">
    Sign in with SSO
  </button>
</form>
```

13. In the Custom CSS code also add this:

```
a.raised.emby-button {
    padding:0.9em 1em;
    color: inherit !important;
}

.disclaimerContainer{
    display: block;
}
```

14. Click **Save** at the bottom & restart the server.
15. When you are signed out you should now see a **Sign in with SSO** button.

:::note
If you have problems check your logs which are under the **Administration** -> **Dashboard** then "logs" and will be near the bottom (most likely) with `Jellyfin.Plugin.SSO_Auth.` as the start of the lines you are looking for.
:::

> ### authentik Configuration

Reviewer: It could be a good idea to start with the authentik config section first, so that client id and secret are available to be used during the jellyfin config directly.
